### PR TITLE
Model apply mission creation

### PR DESCRIPTION
**Pull request by Joelebucheron**
- Missions migration with foreign key
- Mission model creation
- Apply migration with foreign key
- Apply model creation

### CODE REVIEW

File: app/models/mission.rb [hunk 1]
```diff
@@ -1,4 +1,20 @@
 class Mission < ApplicationRecord
-  belongs_to :mentor
-  belongs_to :padawan
+  belongs_to :mentor, class_name: "User"
+  belongs_to :padawan, class_name: "User", optional: true
+
+  enum status: [ :not_started, :started, :finished ]
```
Comment (Vigrouze) [inline app/models/mission.rb]: We think it would be better to update " :started" by "on-going"

File: app/models/mission.rb [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
+  validates :fee, :status, numericality: { only_integer: true }
+  validate end_date_after_start_date
+
+  private
+
+  def end_date_after_start_date
+    return if end_date.blank? || start_date.blank?
+
+    if end_date < start_date
+      errors.add(:end_date, "must be after the start date")
```
Comment (Vigrouze) [inline app/models/mission.rb]: Small detail "Must" vs "must" and "." at the end

**Review by Vigrouze — COMMENTED** [review]
We added two small comments guys ;)

File: app/models/mission.rb (hunk 1 above)
Comment (Joelebucheron) [reply app/models/mission.rb]: Done in next commit

**Pull request merged**